src/main/src/main/java/org/geoserver/GeoServerConfigurationLock.java

```diff
@@ -25,6 +28,10 @@
  * 
  */
 public class GeoServerConfigurationLock {
+    
+    /** DEFAULT_TRY_LOCK_TIMEOUT_MS */
+    private static final int DEFAULT_TRY_LOCK_TIMEOUT_MS = 5000;
```

Comment:
Don't hard code, use GeoServerExtensions.getProperty to allow admins to set this value in multiple ways (even better would be to have a real config option in the global settings, but I understand it's more work and the default should be sane)